dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/messaging/DataRequestedAnsweredEmailMessageSender.kt

```diff
@@ -0,0 +1,107 @@
+package org.dataland.datalandcommunitymanager.services.messaging
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.dataland.datalandbackend.openApiClient.api.CompanyDataControllerApi
+import org.dataland.datalandcommunitymanager.entities.DataRequestEntity
+import org.dataland.datalandcommunitymanager.services.KeycloakUserControllerApiService
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
+import org.dataland.datalandmessagequeueutils.constants.ExchangeName
+import org.dataland.datalandmessagequeueutils.constants.MessageType
+import org.dataland.datalandmessagequeueutils.constants.RoutingKeyNames
+import org.dataland.datalandmessagequeueutils.messages.TemplateEmailMessage
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * Manage sending emails to user regarding data requests
+ */
+@Service("DataRequestedAnsweredEmailSender")
+class DataRequestedAnsweredEmailMessageSender(
+    @Autowired private val cloudEventMessageHandler: CloudEventMessageHandler,
+    @Autowired private val objectMapper: ObjectMapper,
+    @Autowired private val keycloakUserControllerApiService: KeycloakUserControllerApiService,
+    @Autowired private val companyDataControllerApi: CompanyDataControllerApi,
+) {
+    /**
+     * Method to informs user by mail that his request is answered.
+     * @param dataRequestEntity the dataRequestEntity
+     * @param companyName name of the company
+     */
+    fun sendDataRequestedAnsweredEmail(
+        dataRequestEntity: DataRequestEntity,
+        companyName: String = "",
+    ) {
+        val correlationId: String = UUID.randomUUID().toString()
+        val compName = companyName.ifEmpty { getCompanyNameById(dataRequestEntity.datalandCompanyId) }
```

Comment:
1. `val compName` finde ich persönlich ein wenig unschön gewählt, da hier nur versucht wird ein Shadowed name Error zu umgehen.
2. In der Funktion `sendDataRequestedAnsweredEmails` wird bereits versucht die companyId mittels Funktion `getCompanyInfo` zu erhalten. Falls an dieser Stelle die Company bereits leer ist, wird hier bereits ein Fehler auftreten. Wieso wird dann die` .ifEmpty` erst an dieser Stelle verwendet und dann in Kombination mit einer anderen Funktion? Alternativ den Check in die `getCompanyNameById` verschieben und auch nur dort überprüfen?

Reply:
changed